### ISSUE TITLE
QueryBuilder combining 'having' conditions

### ISSUE THREAD

**Issue by mizozobu** [typeorm/typeorm#6801]
**Issue type:**

[ ] bug report
[x] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
It would be nice to have `Brackets` for `having` clause just like `where` clause.
Here is the [commit](https://github.com/typeorm/typeorm/commit/30386f51944556cdea01c535126cd9ba13b669da) that implemented `Brackets` to where(https://github.com/typeorm/typeorm/issues/805).

```typescript
createQueryBuilder("user")
   ...
    .having("user.registered = :registered", { registered: true })
    .andHaving(new Brackets(qb => {
        qb.having("user.firstName = :firstName", { firstName: "Timber" })
          .orHaving("user.lastName = :lastName", { lastName: "Saw" })
    }))
```

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
--!>

**Comment by amiryadid** [comment]
Is there any update on this one? @mizozobu how did you manage eventually?

**Comment by mizozobu** [comment]
I must've got by with raw query.

https://typeorm.io/data-source-api
```ts
const rawData = await dataSource.query(`SELECT * FROM USERS`)
```

https://typeorm.io/entity-manager-api
```ts
const rawData = await manager.query(`SELECT * FROM USERS`)
```

I'm not sure, but probably you can pass parameters to your raw SQL?
https://typeorm.io/select-query-builder#using-parameters-to-escape-data

**Comment by amiryadid** [comment]
Yeah that's what we have today but I wondered maybe you found a better way, shame it doesn't exist as of yet

**Comment by IRCraziestTaxi** [comment]
Was this really never planned? This is something I need.
